cbor/src/main/java/com/fasterxml/jackson/dataformat/cbor/CBORParser.java

```diff
@@ -2549,68 +2549,72 @@ private final String _finishShortText(int len) throws IOException
                     }
-                    final int c3 = inputBuf[inPtr++];
-                    if ((c3 & 0xC0) != 0x080) {
-                        _reportInvalidOther(c3 & 0xFF, inPtr);
+                    break;
+                    case 2:
+                    {
+                        final int c2 = inputBuf[inPtr++];
+                        if ((c2 & 0xC0) != 0x080) {
+                            _reportInvalidOther(c2 & 0xFF, inPtr);
+                        }
+                        final int c3 = inputBuf[inPtr++];
+                        if ((c3 & 0xC0) != 0x080) {
+                            _reportInvalidOther(c3 & 0xFF, inPtr);
+                        }
+                        i = ((i & 0x0F) << 12) | ((c2 & 0x3F) << 6) | (c3 & 0x3F);
                     }
-                    i = ((i & 0x0F) << 12) | ((c2 & 0x3F) << 6) | (c3 & 0x3F);
+                    break;
+                    case 3:
+                        // 30-Jan-2021, tatu: TODO - validate these too?
+                        i = ((i & 0x07) << 18)
+                                | ((inputBuf[inPtr++] & 0x3F) << 12)
+                                | ((inputBuf[inPtr++] & 0x3F) << 6)
+                                | (inputBuf[inPtr++] & 0x3F);
+                        // note: this is the codepoint value; need to split, too
+                        i -= 0x10000;
+                        outBuf[outPtr++] = (char) (0xD800 | (i >> 10));
+                        i = 0xDC00 | (i & 0x3FF);
+                        break;
+                    default: // invalid
+                        _reportInvalidInitial(i);
                 }
-                break;
-            case 3:
-                // 30-Jan-2021, tatu: TODO - validate these too?
-                i = ((i & 0x07) << 18)
-                    | ((inputBuf[inPtr++] & 0x3F) << 12)
-                    | ((inputBuf[inPtr++] & 0x3F) << 6)
-                    | (inputBuf[inPtr++] & 0x3F);
-                // note: this is the codepoint value; need to split, too
-                i -= 0x10000;
-                outBuf[outPtr++] = (char) (0xD800 | (i >> 10));
-                i = 0xDC00 | (i & 0x3FF);
-                break;
-            default: // invalid
-                _reportInvalidInitial(i);
+                outBuf[outPtr++] = (char) i;
+            } while (inPtr < end);
+            String str = _textBuffer.setCurrentAndReturn(outPtr);
+            if (stringRefs != null) {
+                stringRefs.stringRefs.add(str);
+                _sharedString = str;
             }
-            outBuf[outPtr++] = (char) i;
-        } while (inPtr < end);
-        String str = _textBuffer.setCurrentAndReturn(outPtr);
-        if (stringRefs != null) {
-            stringRefs.stringRefs.add(str);
-            _sharedString = str;
+            return str;
+        } catch (RuntimeException re) {
+            throw new IOException("Failed to parse CBOR input", re);
         }
-        return str;
     }
```

Comment:
despite the Github diff - this is just the same code but wrapped in a try-catch

Reply:
Yeah. But I don't think defensive programming makes sense here. I think instead code in question has to check for buffer boundaries and not count on JVM's array checks; the big problem being that there is no information on what in content is broken. (if I remember correctly, having looked that, this is buffer overrun situation; split UTF-8 codepoint or such).

Reply:
* I guess that it is optimal to go and check the bounds instead of catching the exception.
* One advantage to catching the exception is that you don't make the happy path slower by adding additional range checks.
* It does get rid of the annoying public oss-fuzz issue and could be treated as an interim solution until someone gets into the nitty gritty of the code and finds a performant way of checking the array bounds.

Reply:
My concern is mostly that the specific bug could (if I recall correctly) cause corrupt data and NOT hit ArrayIndexOutOfBoundsException, depending on how input is given.

I agree oss-fuzz issue would be nice to resolve, one of the last ones. I fixed a few of these a year or more ago but not this one.